src/agreements/CollateralAgreement.sol

```diff
@@ -98,22 +97,25 @@ contract CollateralAgreementFramework is IAgreementFramework {
         emit AgreementCreated(agreementId, params.termsHash, params.criteria);
     }
 
-    function joinAgreement(uint256 id, uint256 balance) external {
-        if (balance < agreement[id].criteria) revert InsufficientBalance();
-        SafeTransferLib.safeTransferFrom(token, msg.sender, address(this), balance);
+    function joinAgreement(
+        uint256 id,
+        CriteriaResolver calldata resolver
+    ) external override {
+        if (_isPartOfAgreement(id, msg.sender)) revert PartyAlreadyJoined();
+        if (msg.sender != resolver.party) revert PartyMustMatchCriteria();
```

Comment:
```suggestion
        if (_isPartOfAgreement(id, msg.sender))
            revert PartyAlreadyJoined();
        if (msg.sender != resolver.party)
            revert PartyMustMatchCriteria();
```

Reply:
Your suggestions to keep reverts in other line is some kind of style guideline or a personal preference?

Reply:
@xPi2 Just based on the `if` examples at https://docs.soliditylang.org/en/v0.8.15/style-guide.html#control-structures